File: client/src/app/modules/quotation/services/journal.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { JournalType } from '../../quotation/model/JournalType';

@Injectable({
  providedIn: 'root'
})
export class JournalTypeService extends AppRestService<JournalType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getJournalTypes() {
    return this.getListCached(new HttpParams(), "journal-types");
  }

  addOrUpdateJournalType(journalType: JournalType) {
    this.clearListCache(new HttpParams(), "journal-types");
    return this.addOrUpdate(new HttpParams(), "journal-type", journalType, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/JournalTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.JournalType;

public interface JournalTypeService {
    public List<JournalType> getJournalTypes();

    public JournalType getJournalType(Integer id);

    public JournalType addOrUpdateJournalType(JournalType journalType);
}
